lib/content_consistency_checker.rb

```diff
@@ -0,0 +1,76 @@
+require 'gds_api/content_store'
+
+class ContentConsistencyChecker
+  attr_reader :errors
+
+  def initialize(content_id, locale = "en")
+    @content_id = content_id
+    @locale = locale
+    @errors = []
+  end
+
+  def call
+    unless document
+      errors << "Document (#{content_id}, #{locale}) could not be found."
+      return errors
+    end
+
+    check_edition(draft, draft_content_store) if draft
+    check_edition(live, live_content_store) if live
+
+    errors
+  end
+
+private
+
+  attr_reader :content_id, :locale
+
+  def item_from_content_store(path, content_store)
+    begin
+      content_store.content_item(path).parsed_content
+    rescue GdsApi::ContentStore::ItemNotFound, GdsApi::HTTPGone, GdsApi::HTTPForbidden
+      nil
+    end
+  end
+
+  def check_edition(edition, content_store)
+    content_item = item_from_content_store(edition.base_path, content_store)
+
+    if edition.gone? && content_item
+      errors << "Content exists in the content store."
+    elsif content_item.nil?
+      errors << "Content is missing from the content store."
+      return
+    end
+
+    fields = [:rendering_app, :publishing_app, :schema_name, :document_type]
```

Comment:
Are we able to check all the fields that content_store makes available? just so this can be a more general sync-check with content_store? 

An awkward field is links as you'd need to run the edition through the downstream presenter and would probably make this run way slower - but then that's probably the most useful field for a check, what do you think?